### DISCUSSION TITLE
Recommended esp32 board?

### DISCUSSION THREAD

**Discussion by macca16** [krzmaz/LinakDeskEsp32Controller#10] [Q&A] [answered]
Hi, is there a recommended esp32 board? I managed to flash an m5stack without issue but running into a few issues with the NimBLE lib throwing after config is successfully parsed and when servicing an incoming http request. Likely unrelated to the hardware but would be good to know of 'good' known hardware.

Thanks

**Comment by krzmaz** [comment, marked as answer]
Hi, the board manufacturer should not matter, as long as the important parameters like memory are the same.
You need to look out for the boards that use PSRAM, in which case you might need to add
```
    -DBOARD_HAS_PSRAM
    -mfix-esp32-psram-cache-issue
```
to `build_flags` - see https://github.com/TheRealMazur/LinakDeskEsp32Controller/issues/7

Other than the memory issue, the only difference I saw was the WiFi/BT signal strength differing between boards.
I had the best experience with a board like this one: [link](https://www.amazon.com/ACEIRMC-ESP-WROOM-32-Bluetooth-Development-Compatible/dp/B08L5X2ZM3/) - link is **not** affiliate 😁

**Reply by macca16** [reply]
Thank you for the info. #7 wasn't the issue I was having but I increased the logging and determined that my phone having an active connection meant that the desk was not entering pairing mode properly. Now that I have rectified that I've been able to connect to the desk (having obtained the Mac Address using the Android version of nRF Connect, as the iOS version doesn't show MAC Addresses for some reason) and get the web interface up without a crash and am able to call the /getHeight endpoint successfully.

Many thanks for your help and this project!

**Reply by krzmaz** [reply]
Glad to hear it's working for you! 
Feel free to create PRs/Issues for any missing documentation and features. This repository has started as a documentation of my work and it is full of assumptions about the users knowledge, as at the moment of creating it, I was the only user 😁
